### PR TITLE
fix(selfdestruct): invoke selfDestructAtTransactionEnd

### LINKED ISSUE #1017: Updating deployment information at the point of effective SELFDESTRUCT
The `DeploymentInfo` class needs to implement a method
```java
  public void selfdestructAtTransactionEnd(Address address) {
    this.incrementDeploymentNumber(address);
    this.unmarkDeploying(address);
  }
```
Which should get called on the account-row where `MARKED_FOR_SELFDESTRUCT` goes from 0 to 1.

### PR DESCRIPTION
**Pull request by lorenzogentile404**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java [hunk 1]
```diff
@@ -275,6 +274,11 @@ public void resolvePostTransaction(
                   accountBeforeSelfDestruct,
                   selfdestructorAccountAfter.wipe(),
                   DomSubStampsSubFragment.selfdestructDomSubStamps(hub));
+
+      // TODO: ensure this is the right place for this call
+      final DeploymentInfo deploymentInfo = hub.transients().conflation().deploymentInfo();
+      deploymentInfo.updateDeploymentInfoForSelfdestructsAtTransactionEnd(this.recipientAddress);
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java]: The update must happen at transaction end, and this is done in the `resolvePostTransaction` method. So this should be fine. This `updateDeploymentInfoForSelfdestructsAtTransactionEnd` method was renamed in my latest PR.

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/DeploymentInfo.java [hunk 2]
```diff
@@ -61,6 +60,11 @@ private int getDeploymentNumber(Address address) {
     return this.deploymentNumber.getOrDefault(address, 0);
   }
 
+  public void deploy(Address address) {
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/DeploymentInfo.java]: This already existed, didn't it ? I had this method on my branch, too. I'm confused.

**Comment by OlivierBBB** [comment]
I'm not sure we want to merge this PR. Unless I'm mistaken every part of it has been updated in the mean time except for comment deletions.

**Comment by lorenzogentile404** [comment]
@OlivierBBB shall I apply the same edits to the latest `arith-dev` instead of merging this?

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/DeploymentInfo.java (hunk 2 above)
Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/DeploymentInfo.java]: It was part of the conflicts I solved during a merge of `arith-dev `into this branch. Probably there were two versions and I kept the one of `arith-dev`.

**Comment by OlivierBBB** [comment]
@lorenzogentile404 We should take a few minutes today to review what we need.

**Review by OlivierBBB — APPROVED** [review]
LGTM

**Pull request merged**